projects/npm-tools/packages/npm-scripts/src/scripts/build.js

```diff
@@ -19,6 +18,7 @@ const validateConfig = require('../utils/validateConfig');
 const webpack = require('./webpack');
 
 const BUILD_CONFIG = getMergedConfig('npmscripts', 'build');
+const FEDERATION_ENABLED = getMergedConfig('npmscripts', 'federation');
```

Comment:
Thing about calling `getMergedConfig` twice, both times with a second param, is that it will throw if the key is not supplied. It seems fine for `build` to be required, but `federation` might be better as an optional key, seeing as absent is the same as `false`. (Especially because in DXP there are modules that don't use the preset, so you'll have to add an explicit `federation` key to all of them, or to the alternative preset that they do use.)

So maybe:

```js
const CONFIG = getMergedConfig('npmscripts');

if (!CONFIG.build) {
  throw new Error('npmscripts.config.js is missing required "build" key');
}
```

and from then on just use `CONFIG.build` and `CONFIG.federation`.